orchagent/aclorch.cpp

```diff
@@ -3848,6 +4028,35 @@ bool AclOrch::updateAclTable(AclTable &currentTable, AclTable &newTable)
     return true;
 }
 
+bool AclOrch::removeEgrSetDscpTable(string table_id)
+{
+    m_egrSetDscpRef.erase(table_id);
+    if (m_egrSetDscpRef.size() == 0)
+    {
+        if (!removeAclTable(EGR_SET_DSCP_TABLE_ID))
+        {
+            SWSS_LOG_ERROR("Failed to remove ACL table %s", EGR_SET_DSCP_TABLE_ID);
+            return false;
+        }
+    }
+    else
+    {
+        //create a dummy table with no ports. The updateAclTable will remove the ports
+        // which were associated with the egrSetDscpTable we just added because the
```

Comment:
What's the purpose for this dummy table?

Reply:
Since we only create one instance of EGR_SET_DSCP, the bind ports it needs to support have to be a set of  MARK_META and Mark_METAV6. if e.g. MARK_META hs bind ports [Eth1, Eth2] and MARK_METAV6 has [Eth3,Eth4] then EGR_SET_DSCP needs to bind on Eth1, Eth2, Eth3, Eth4. if one of the table is removed we need to remove its ports form EGR_SET_DSCP. This EGR_SET_DSCP dummy table object  is created if one instance of the MARK_META table is removed. we take its bind ports and take those out of the actual EGR_SET_DSCP table.

Reply:
Where is the port merge happened?

Reply:
@siqbal1986 Can you check this comment?